PlaygroundService/Controllers/PlaygroundController.cs

```diff
@@ -86,29 +93,193 @@ public async Task<IActionResult> BuildService(IFormFile contractFiles)
                 var dllBytes = BytesExtension.Read(pathToDll);
                 if (dllBytes == null)
                 {
-                    _logger.LogError("PlaygroundController - BuildProject method returned error: dllBytes is null " + message  + " time:" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
+                    _logger.LogError("PlaygroundController - BuildProject method returned error: dllBytes is null " + message + " time:" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
                     return BadRequest(new PlaygroundSchema.PlaygroundContractGenerateResponse
                     {
                         Success = false,
                         Message = "Error in dll."
                     });
                 }
-                
+
                 var res = Content(Convert.ToBase64String(dllBytes));
-                
+
                 _logger.LogInformation("PlaygroundController - BuildProject method over: " + " time:" + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
                 var endTime = DateTime.Now;
-                
+
                 _logger.LogInformation("PlaygroundController - BuildProject method took: " + (endTime - startTime).TotalSeconds + " seconds");
                 return res;
             }
-            
+
             _logger.LogError("PlaygroundController - BuildProject method returned error: " + message);
             return BadRequest(new PlaygroundSchema.PlaygroundContractGenerateResponse
             {
                 Success = success,
                 Message = message
             });
         }
+
+        [HttpPost("share/create")]
+        public async Task<IActionResult> UploadZipFile(IFormFile file)
+        {
+            // Check if the file is a valid ZIP file
+            if (file == null || file.Length == 0 || !file.FileName.EndsWith(".zip"))
+            {
+                return BadRequest("Invalid file. Only ZIP files are allowed.");
+            }
+            // Check file size
+            if (file.Length > _maxFileSizeInBytes)
+            {
+                return BadRequest($"File size exceeds the {_maxFileSizeInBytes / (1024 * 1024)} MB limit.");
+            }
+
+            // Save the file temporarily to scan for viruses
+            var tempFilePath = Path.GetTempFileName();
```

Comment:
Can you create a grain for the logic from this point on instead? Generally, controller should only be doing dto conversion and input validation. The grain will do the actual operation.